serverless/src/utils/toLegacyJSON.js

```diff
@@ -0,0 +1,148 @@
+import { XMLParser } from 'fast-xml-parser'
+
+import getConceptSchemes from '@/getConceptSchemes/handler'
+import { getSkosConcept } from '@/shared/getSkosConcept'
+
+// Helper function to retrieve parsed information in concept_schemes xml
+function findMatchingLongName(schemeShortName, parsedData) {
+  const schemes = parsedData.schemes.scheme
+
+  const matchingScheme = schemes.find(
+    (scheme) => scheme['@name'] === schemeShortName
+  )
+
+  return matchingScheme['@longName']
+}
+
+// Helper function to determine if there are multiple altLabels and assist in translating the different types
+function processAltLabels(altLabels) {
+  if (!altLabels) {
+    return []
+  }
+
+  const labelArray = Array.isArray(altLabels) ? altLabels : [altLabels]
+
+  return labelArray.map((label) => {
+    const processedLabel = {}
+
+    if (label['@gcmd:category']) {
+      processedLabel.category = label['@gcmd:category']
+    }
+
+    processedLabel.text = label['@gcmd:text']
+
+    return processedLabel
+  })
+}
+
+const toLegacyJSON = async (conceptIRI, skosConcept) => {
```

Comment:
Overall this looks really good.  I think the only thing that could be improved is the and we'll need in order to make this fast enough is make toLegacyJson not have to do any async calls, that meams there can't be any await calls in it.   All the information it needs shoudl be passed into the function.

I think to solve this, you just need to pass in 2 maps to this the toLegacyJSON function:
conceptSchemeMap[schemeShortName] and returns the long name.
conceptPrefLabelMap[uuid] and returns the prefLabel.

Then you can use these maps instead of calling await to get this information.  From what I can tell this should work, with the exception of isLeaf for broader, narrower, and in this case, we lets just not include this.